### DISCUSSION TITLE
关于组件是怎么画出来的

### DISCUSSION THREAD

**Discussion by icuxika** [vczh/GacUIBlog#12] [General]
轮子哥，你好，请问能够介绍一下，最上层的Button、Label等组件在各种布局下，是怎么经过转换，最后在窗口上画出来的吗？学过一点OpenGL之类的，像上层的ui是经过计算得出一份顶点数据然后渲染出来吗

**Comment by vczh** [comment]
布局跟渲染是互相独立的，我在计算完布局之后，会把所有的坐标都存下来，供渲染的时候使用。
